### PR TITLE
fix flexible route scenario for routes

### PR DESCRIPTION
**Pull request by awmackowiak**

### CODE REVIEW

File: vaas-app/src/vaas/vcl/templates/vcl_blocks/4.0/RECV.tvcl [hunk 1]
```diff
@@ -6,6 +6,11 @@ sub vcl_recv {
     # Call protocol redirect sub
     call protocol_redirect;
 
+    # Handler for no backend in director
+    if(req.http.x-action == "nobackend") {
+        return(synth(404, "<!--Director " + req.http.x-director + " has no backends or is disabled-->"));
```
Comment (ziollek) [inline vaas-app/src/vaas/vcl/templates/vcl_blocks/4.0/RECV.tvcl:11]: I suppose after that change - replacing tag <SET_BACKEND_*> in renderer.py#344 is uneccessary

**Review by ziollek — CHANGES_REQUESTED** [review]
I think its good time to briefly refresh our docs regarding to vcl templates https://vaas.readthedocs.io/en/latest/documentation/vcl/

File: vaas-app/src/vaas/vcl/templates/vcl_blocks/4.0/RECV.tvcl (hunk 1 above)
Comment (awmackowiak) [reply vaas-app/src/vaas/vcl/templates/vcl_blocks/4.0/RECV.tvcl:11]: that's true. I will change this.

**Review by smigacz — APPROVED** [review]

**Review by ziollek — APPROVED** [review]

File: docs/documentation/vcl.md [hunk 2]
```diff
@@ -20,6 +20,7 @@ The default VCL generated by VaaS has the following structure:
                 <BACKEND_DEFINITION_LIST_{DIRECTOR}_{DC}/>
                 <DIRECTOR_DEFINITION__{DIRECTOR}_{DC}/>
             </DIRECTOR_{DIRECTOR}>
```
Comment (ziollek) [inline docs/documentation/vcl.md:22]: <DIRECTOR_INIT_{DIRECTOR}/>

**Review by ziollek — APPROVED** [review]

**Review by TheMickeyMike — APPROVED** [review]

**Review by ziollek — APPROVED** [review]

**Pull request merged**